PathPlanning/RRTStarReedsShepp/rrt_star_reeds_shepp.py

```diff
@@ -166,6 +167,28 @@ def steer(self, from_node, to_node):
 
         return new_node
 
+    def rewire(self, new_node, near_inds):
+        for i in near_inds:
+            near_node = self.node_list[i]
+            edge_node = self.steer(new_node, near_node)
+            if not edge_node:
+                continue
+            edge_node.cost = self.calc_new_cost(new_node, near_node)
+
+            no_collision = self.check_collision(
+                edge_node, self.obstacle_list, self.robot_radius)
+            improved_cost = near_node.cost > edge_node.cost
+
+            if no_collision and improved_cost:
+                near_node.x = edge_node.x
```

Comment:
I don’t remember this code well, but just
> near_node = edge_node

does not work?

Reply:
A simple example is written to explain why near.x=edge.x instead of near = edge here.
Please check this below:

<img width="982" alt="Snipaste_2023-04-03_11-02-41" src="https://user-images.githubusercontent.com/39330401/229402976-71a2ea0f-ac05-477f-b5e7-0a79014a34fa.png">

But I think you're right, there are indeed cleaner ways to do the replacements:

`self.node_list[i] = edge_node`

which ignores the intermediate variable 'near_node' and directly replaces the i-th element in the node list.

I updated this part in my new commit, thank you for pointing it out 👍

Reply:
A similar situation occurs in rewire() of rrt_star.py, so I also made a change there.

Reply:
self.propagate_cost_to_leaves() is taken out from the for loop of rewire() because it already contains an internal loop for iterating through all nodes in the node list. It's OK to just run it once after all rewires.